src/tests/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 find_package(Boost 1.39.0 REQUIRED COMPONENTS unit_test_framework)
-foreach(PROG test_elements test_edge test_name test_typeconverter )
-  file(GLOB ${PROG}_SOURCES ${PROG}*.cc)
+foreach(PROG test_elements test_edge test_name test_typeconverter test_graphnode test_edgecontainer)
+  file(GLOB ${PROG}_SOURCES ${PROG}.cc)
```

Comment:
If you drop the `*` the whole globbing command can dropped! It would also disallow for a test using multiple files!

Reply:
It might be better to rename `test_edge` to `test_edge_simple`.

Reply:
Fair enough, but why do we need the GLOB anyway if we are going to be explicitly writing out all the tests?

Reply:
Well you copied the loop from `csg`. And there the idea was that all files which start with a certain pattern get compiled into one executable, e.g. `csg_stat*.cc` matches `csg_stat.cc` and `csg_stat_imc.cc` and both files get compiled into the `csg_stat` executable. Downside is of course, that the program prefix has to be unique, which `test_edge` isn't, when having files like `test_edge_container.cc`.

I guess the globbing would make sense here, too, if you don't want to put all e.g. edge-related test in one file.